File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.cpp
class Solution {
public:
    int missingInteger(vector<int>& nums) {
        int n = nums.size();
        int seqSum = nums[0];

        for(int i=1;i<n;i++){
            if(nums[i] == nums[i-1] +1)
                seqSum += nums[i];
            else
                break;
        }

        vector<bool>hash(1276, false);

        for(int &num: nums) hash[num] = true;

        while(hash[seqSum])
            seqSum++;

        return seqSum;

    }
};

File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.ts
function missingInteger(nums: number[]): number {
    let seqSum = nums[0];

    for(let i=1;i<nums.length;i++){
        if(nums[i] === nums[i -1] +1)
            seqSum += nums[i];
        else
            break;
    }

    const present = new Array(1276).fill(false);

    for(const num of nums) present[num] = true;

    while(present[seqSum])
        seqSum++;

    return seqSum;

};